variables.tf

```diff
@@ -29,13 +29,16 @@ variable "client_dns_server_ips" {
 }
 
 variable "client_auth_methods" {
-  type        = string
-  description = "Client authentication method"
-  default     = "username"
+  type        = list(string)
+  description = "The methods used to authenticate VPN clients to this VPN server. Allowable values are: certificate, username. For more information, see https://cloud.ibm.com/docs/vpc?topic=vpc-vpn-client-environment-setup"
+  default     = ["username"]
+  validation {
+    error_message = "Allowed values are username and certificate."
+    condition     = alltrue([for method in var.client_auth_methods : contains(["username", "certificate"], method)])
+  }
   validation {
-    error_message = "Only authentication by username is supported."
-    condition = can(contains(["username"], var.client_auth_methods)
-    )
+    error_message = "Each value (username or certificate) may appear at most once (no duplicates allowed)"
```

Comment:
username and certificate are mutually exclusive in the authentication block. You can have N number of authentication block for each user.

I think with your logic, you can't have more than 2 elements in the list, as there are only 2 allowed values - username and certification. 

I would remove this logic.

Reply:
with `client_auth_methods` input variable you define the authentication `type`, we support only 2: `username` or `certificate` or `both`.. it means max elements can be `2`, and we do not allow duplications such as `[username, username]`.

Reply:
Thanks for the clarification, looks good